pyrmq/publisher.py

```diff
@@ -56,19 +64,25 @@ def __init__(self, exchange_name: str, queue_name: str, routing_key: str, **kwar
         self.exchange_name = exchange_name
         self.queue_name = queue_name
         self.routing_key = routing_key
+        self.exchange_type = exchange_type
         self.host = kwargs.get("host") or os.getenv("RABBITMQ_HOST") or "localhost"
         self.port = kwargs.get("port") or os.getenv("RABBITMQ_PORT") or 5672
-        self.username = kwargs.get("username") or "guest"
-        self.password = kwargs.get("password") or "guest"
-        self.connection_attempts = kwargs.get("connection_attempts") or 3
-        self.retry_delay = kwargs.get("retry_delay") or 5
-        self.retry_backoff_base = kwargs.get("retry_backoff_base") or 2
-        self.retry_backoff_constant_secs = (
-            kwargs.get("retry_backoff_constant_secs") or 5
-        )
+        self.username = kwargs.get("username", "guest")
+        self.password = kwargs.get("password", "guest")
+        self.connection_attempts = kwargs.get("connection_attempts", 3)
+        self.retry_delay = kwargs.get("retry_delay", 5)
+        self.retry_backoff_base = kwargs.get("retry_backoff_base", 2)
+        self.retry_backoff_constant_secs = kwargs.get("retry_backoff_constant_secs", 5)
         self.error_callback = kwargs.get("error_callback")
-        self.infinite_retry = kwargs.get("infinite_retry") or False
-        self.queue_args = kwargs.get("queue_args") or None
+        self.infinite_retry = kwargs.get("infinite_retry", False)
+        self.queue_args = kwargs.get("queue_args")
+
+        if self.exchange_type == "direct" and (
```

Comment:
Assign `direct` to a constant

Reply:
Hmm. Waiting on https://github.com/pika/pika/pull/1260. Will update to use this once merged.